src/core/src/op/transpose.cpp

```diff
@@ -25,74 +27,60 @@ bool ngraph::op::v1::Transpose::visit_attributes(AttributeVisitor& visitor) {
-        }
-        NODE_VALIDATION_CHECK(this,
-                              is_valid_permutation(permutation, arg_shape.rank()),
-                              "Permutation ",
-                              permutation,
-                              " is not valid for input shape ",
-                              arg_shape);
-        set_output_type(0, get_input_element_type(0), ngraph::apply_permutation(arg_shape, permutation));
-    } else {
-        Rank output_rank = arg_shape.rank();
-        if (output_rank.is_dynamic() && input_order_shape.is_static() && input_order_shape[0].get_length())
-            output_rank = input_order_shape[0];
-        set_output_type(0, get_input_element_type(0), ov::PartialShape::dynamic(output_rank));
-    }
-    NGRAPH_SUPPRESS_DEPRECATED_END
+    set_input_is_relevant_to_shape(ORDER);
+
+    std::vector<ov::PartialShape> input_shapes{arg_shape, input_order_shape};
+    std::vector<ov::PartialShape> output_shapes(OUT_COUNT, ov::PartialShape{});
+
+    shape_infer(this, input_shapes, output_shapes);
+
+    set_output_size(output_shapes.size());
+    set_output_type(ARG, get_input_element_type(ARG), output_shapes[ARG_T]);
 }
 
 shared_ptr<Node> op::v1::Transpose::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v1_Transpose_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return make_shared<v1::Transpose>(new_args[0], new_args[1]);
+    return make_shared<v1::Transpose>(new_args[ARG], new_args[ORDER]);
 }
 
 namespace transpose {
 namespace {
+using namespace ov::op;
+
 bool evaluate_transpose(const HostTensorPtr& arg1, const HostTensorPtr& arg2, const HostTensorPtr& out) {
-    NGRAPH_CHECK(arg2->get_element_type().is_integral_number(),
-                 "Transpose axis element type has to be integral data type.");
+    OPENVINO_ASSERT(arg2->get_element_type().is_integral_number(),
+                    "Transpose axis element type has to be integral data type.");
 
     std::vector<int64_t> axes_order = host_tensor_2_vector<int64_t>(arg2);
     ov::Shape in_shape = arg1->get_shape();
     if (shape_size(arg2->get_shape()) == 0) {
-        axes_order.resize(in_shape.size());
-        std::iota(axes_order.begin(), axes_order.end(), 0);
-        std::reverse(axes_order.begin(), axes_order.end());
+        ov::generate_transpose_default_order(axes_order, in_shape.size());
     } else {
-        std::unordered_set<int64_t> axes_set(axes_order.begin(), axes_order.end());
-        bool is_unique_order = axes_set.size() == axes_order.size();
-        NGRAPH_CHECK(is_unique_order, "Transpose axes order values must be unique.");
+        OPENVINO_ASSERT(ov::is_valid_axes_order(axes_order, in_shape.size()),
+                        "Permutation ",
+                        AxisVector(axes_order.begin(), axes_order.end()),
+                        " is not valid for input shape ",
+                        in_shape);
```

Comment:
lets call shape_infer here? seems like duplication of shape inference

Reply:
Refactored.